php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1532 @@
+        b.optional(
+          f.newTuple92(
+            b.token(PHPPunctuator.EQU),
+            STATIC_SCALAR())))
+    );
+  }
+
+  public SeparatedList<NamespaceNameTree> INTERFACE_LIST() {
+    return b.<SeparatedList<NamespaceNameTree>>nonterminal(PHPLexicalGrammar.INTERFACE_LIST).is(
+      f.interfaceList(
+        NAMESPACE_NAME(),
+        b.zeroOrMore(f.newTuple98(b.token(COMMA), NAMESPACE_NAME()))));
+  }
+
+  public TraitUseStatementTree TRAIT_USE_STATEMENT() {
+    return b.<TraitUseStatementTree>nonterminal(PHPLexicalGrammar.TRAIT_USE_STATEMENT).is(
+      b.firstOf(
+        f.traitUseStatement(
+          b.token(PHPKeyword.USE),
+          INTERFACE_LIST(),
+          EOS()),
+        f.traitUseStatement(
+          b.token(PHPKeyword.USE),
+          INTERFACE_LIST(),
+          b.token(LCURLYBRACE),
+          b.zeroOrMore(
+            b.firstOf(
+              TRAIT_PRECEDENCE(),
+              TRAIT_ALIAS())),
+          b.token(RCURLYBRACE))));
+  }
+
+  public TraitPrecedenceTree TRAIT_PRECEDENCE() {
+    return b.<TraitPrecedenceTree>nonterminal(PHPLexicalGrammar.TRAIT_PRECEDENCE).is(
+      f.traitPrecedence(
+        TRAIT_METHOD_REFERENCE_FULLY_QUALIFIED(),
+        b.token(PHPKeyword.INSTEADOF),
+        INTERFACE_LIST(),
+        EOS()));
+  }
+
+  public TraitAliasTree TRAIT_ALIAS() {
+    return b.<TraitAliasTree>nonterminal(PHPLexicalGrammar.TRAIT_ALIAS).is(
+      b.firstOf(
+        f.traitAlias(
+          TRAIT_METHOD_REFERENCE(),
+          b.token(PHPKeyword.AS),
+          b.optional(MEMBER_MODIFIER()),
+          IDENTIFIER(),
+          EOS()),
+        f.traitAlias(
+          TRAIT_METHOD_REFERENCE(),
+          b.token(PHPKeyword.AS),
+          MEMBER_MODIFIER(),
+          EOS())
+      ));
+  }
+
+  public TraitMethodReferenceTree TRAIT_METHOD_REFERENCE() {
+    return b.<TraitMethodReferenceTree>nonterminal(PHPLexicalGrammar.TRAIT_METHOD_REFERENCE).is(
+      b.firstOf(
+        TRAIT_METHOD_REFERENCE_FULLY_QUALIFIED(),
+        f.traitMethodReference(
+          b.token(PHPLexicalGrammar.IDENTIFIER))));
+  }
+
+  public TraitMethodReferenceTree TRAIT_METHOD_REFERENCE_FULLY_QUALIFIED() {
+    return b.<TraitMethodReferenceTree>nonterminal(PHPLexicalGrammar.TRAIT_METHOD_REFERENCE_FULLY_QUALIFIED).is(
+      f.traitMethodReference(
+        NAMESPACE_NAME(),
+        b.token(PHPPunctuator.DOUBLECOLON),
+        b.token(PHPLexicalGrammar.IDENTIFIER)));
+  }
+
+  /**
+   * [ END ] Declaration
+   */
+
+  /**
+   * [ START ] Statement
+   */
+
+  public StatementTree TOP_STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.TOP_STATEMENT)
+      .is(b.firstOf(
+        CLASS_DECLARATION(),
+        TRAIT_DECLARATION(),
+        FUNCTION_DECLARATION(),
+        INTERFACE_DECLARATION(),
+        NAMESPACE_STATEMENT(),
+        USE_STATEMENT(),
+        CONSTANT_DECLARATION(),
+        STATEMENT()
+      ));
+  }
+
+  public NamespaceStatementTree NAMESPACE_STATEMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)